src/main/java/com/minecolonies/coremod/colony/managers/BuildingManager.java

```diff
@@ -506,8 +507,19 @@ public IBuilding addNewBuilding(@NotNull final AbstractTileEntityColonyBuilding
     @Override
     public void removeBuilding(@NotNull final IBuilding building, final Set<ServerPlayerEntity> subscribers)
     {
-        if (buildings.remove(building.getID()) != null)
+        if (buildings.containsKey(building.getID()))
         {
+            final ImmutableMap.Builder<BlockPos, IBuilding> builder = new ImmutableMap.Builder<>();
+            for (final IBuilding tbuilding : buildings.values())
+            {
+                if (tbuilding != building)
```

Comment:
Shouldn't this be comparing the building ids instead?  It might be possible to not have reference equality...

Reply:
if we do not have reference equality here then we have a huge error elsewhere, as this very map should be the one holding all existing references